File: frontend/lib/models/work_attestation.dart
import 'user.dart';
import 'work.dart';
import 'equipment_type.dart';

class WorkAttestation {
  final String id;
  final String manoeuverSheetId;
  final Work work;
  final User worker;
  final List<EquipmentType> equipmentTypes;
  final DateTime createdAt;
  final DateTime updatedAt;

  WorkAttestation({
    required this.id,
    required this.manoeuverSheetId,
    required this.work,
    required this.worker,
    required this.equipmentTypes,
    required this.createdAt,
    required this.updatedAt,
  });

  factory WorkAttestation.fromJson(Map<String, dynamic> json) {
    return WorkAttestation(
      id: json['id'],
      manoeuverSheetId: json['manoeuverSheet']['id'],
      work: Work.fromJson(json['work']),
      worker: User.fromJson(json['worker']),
      equipmentTypes: (json['equipmentTypes'] as List)
          .map((type) => EquipmentType.fromString(type.toString()) ?? EquipmentType.TRANSFORMATEUR)
          .toList(),
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'manoeuverSheetId': manoeuverSheetId,
      'work': work.toJson(),
      'worker': worker.toJson(),
      'equipmentTypes': equipmentTypes.map((type) => type.name).toList(),
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
} 

File: backend/src/main/java/com/steg/loto/backend/models/WorkAttestation.java
package com.steg.loto.backend.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.util.UUID;
import java.util.List;

@Entity
@Table(name = "work_attestations")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class WorkAttestation {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "manoeuver_sheet_id", nullable = false)
    private ManoeuverSheet manoeuverSheet;

    @ManyToOne
    @JoinColumn(name = "work_id", nullable = false)
    private Work work;

    @ManyToOne
    @JoinColumn(name = "worker_id", nullable = false)
    private User worker;

    @ElementCollection
    @CollectionTable(name = "work_attestation_equipment_types")
    @Column(name = "equipment_type")
    private List<String> equipmentTypes;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
} 